jobs-forwarder/internal/forwarder/baseforwarder.go

```
@@ -0,0 +1,107 @@
+package forwarder
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type BaseForwarder struct {
+	ctx        context.Context
+	g          *errgroup.Group
+	jobsDB     jobsdb.JobsDB
+	log        logger.Logger
+	config     *config.Config
+	baseConfig struct {
+		pickupSize                int
+		loopSleepTime             time.Duration
+		jobsDBQueryRequestTimeout time.Duration
+		jobsDBMaxRetries          int
+	}
+}
+
+func (bf *BaseForwarder) LoadMetaData(ctx context.Context, g *errgroup.Group, schemaDB jobsdb.JobsDB, log logger.Logger, config *config.Config) {
+	bf.config = config
+	bf.baseConfig.pickupSize = bf.config.GetInt("JobsForwarder.eventCount", 10000)
+	bf.baseConfig.loopSleepTime = config.GetDuration("JobsForwarder.loopSleepTime", 10, time.Second)
+	bf.baseConfig.jobsDBQueryRequestTimeout = config.GetDuration("JobsForwarder.queryTimeout", 10, time.Second)
+	bf.baseConfig.jobsDBMaxRetries = config.GetInt("JobsForwarder.maxRetries", 3)
+	bf.log = log
+	bf.jobsDB = schemaDB
+	bf.ctx = ctx
+	bf.g = g
+}
+
+func (bf *BaseForwarder) GetJobs(ctx context.Context) ([]*jobsdb.JobT, bool, error) {
+	if bf.jobsDB == nil {
+		bf.log.Errorf("base forwarder: jobsDB is nil")
+		return []*jobsdb.JobT{}, true, nil
+	}
```

I believe that `jobsDB` should never be nil, let it panic
```suggestion
```